### PR TITLE
[Bug Fix]Zotero/Mendeley fix rework

### PR DESCRIPTION
**Pull request by GageGaskins**
# Purpose
Redo Zotero/Mendeley API fix so that project settings nodes cannot be used if credentials are wrong. User must now to go User Settings and fix before they can be used again. Took out growl box from project page.

# Side Effects
Disables node use in project settings until error is fixed.

### CODE REVIEW

File: website/addons/citations/provider.py [hunk 1]
```diff
@@ -17,6 +17,14 @@ def __init__(self, provider_name):
     def serializer(self):
         pass
 
+    def check_credentials(self, node_addon):
+        valid = True
+        try:
+            node_addon.api.client()
+        except HTTPError:
```
Comment (sloria) [inline website/addons/citations/provider.py]: This error handling can be improved; only certain status codes should be allowed to pass here (probably 403).  Other status codes should still raise an error.

File: website/addons/citations/provider.py [hunk 2]
```diff
@@ -17,6 +17,14 @@ def __init__(self, provider_name):
     def serializer(self):
         pass
 
+    def check_credentials(self, node_addon):
+        valid = True
+        try:
+            node_addon.api.client()
```
Comment (sloria) [inline website/addons/citations/provider.py]: Is this correct? `client` appears to be a property of the `Mendeley` provider class, not a method.

File: website/addons/citations/provider.py [hunk 3]
```diff
@@ -17,6 +17,14 @@ def __init__(self, provider_name):
     def serializer(self):
         pass
 
+    def check_credentials(self, node_addon):
```
Comment (sloria) [inline website/addons/citations/provider.py]: Please write a unit test for this method.

**Comment by sloria** [comment]
Pass finished.

**Comment by sloria** [comment]
@GageGaskins Is this a fix for https://github.com/CenterForOpenScience/osf.io/pull/4477/files ?  If so, please resend this to `release/0.59.0`. Also be sure to remove any redundant code. For example, these lines: https://github.com/CenterForOpenScience/osf.io/blob/ae2b5d8bf49b2871a4f4b37344b0b3350e467e21/website/addons/mendeley/model.py#L76-L88 seem to do what you are doing `check_credentials`.

**Comment by GageGaskins** [comment]
Correct branch/pull request at #4657

**Pull request closed without merging**